Reports/router.js

```diff
@@ -1,4 +1,38 @@
 var express = require('express');
+var mongoose = require('mongoose');
```

Comment:
I agree.  Most of this code was written just to check that the models were correct.  These methods will most likely be pulled into a separate file in a future PR.

I'm not sure what you mean by a base implementation for MongoDB-specific calls, since all DB object interaction is done through the model objects.